vic/drivers/image/src/get_global_domain.c

```diff
@@ -66,95 +68,95 @@ get_global_domain(char          *nc_name,
 
     for (y = 0, i = 0; y < global_domain->n_ny; y++) {
         for (x = 0; x < global_domain->n_nx; x++, i++) {
-            if (run[i] == 1) {
-                global_domain->ncells++;
+            if (run[i] == true) {
+                global_domain->ncells_active++;
             }
         }
     }
 
     // if MASTER_PROC
-    global_domain->locations = (location_struct *)
-                               malloc(global_domain->ncells *
-                                      sizeof(location_struct));
+    global_domain->locations = malloc(global_domain->n_ny * global_domain->n_nx *
+                                      sizeof(*global_domain->locations));
     if (global_domain->locations == NULL) {
         log_err("Memory allocation error in get_global_domain().");
     }
-    for (i = 0; i < global_domain->ncells; i++) {
+    for (i = 0; i < global_domain->ncells_total; i++) {
         initialize_location(&(global_domain->locations[i]));
     }
 
     // allocate memory for variables
-    var = (double *) malloc(global_domain->n_ny * global_domain->n_nx *
-                            sizeof(double));
+    var = malloc(global_domain->ncells_total *
+                 sizeof(*var));
     if (var == NULL) {
         log_err("Memory allocation error in get_global_domain().");
     }
 
     for (y = 0, i = 0, j = 0; y < global_domain->n_ny; y++) {
         for (x = 0; x < global_domain->n_nx; x++, i++) {
-            if (run[i] == 1) {
-                global_domain->locations[j].io_idx = i;
-                global_domain->locations[j].global_idx = j;
-                j++;
+            if (run[i] == true) {
+                global_domain->locations[i].run = 1;
             }
         }
     }
 
-    // get 1D indices used in mapping the netcdf fields to the locations
-    idx = (size_t *) malloc(global_domain->ncells *
-                            sizeof(size_t));
-    if (idx == NULL) {
-        log_err("Memory allocation error in vic_init().");
-    }
-    for (i = 0; i < global_domain->ncells; i++) {
-        idx[i] = global_domain->locations[i].io_idx;
+    for (y = 0, i = 0, j = 0; y < global_domain->n_ny; y++) {
+        for (x = 0; x < global_domain->n_nx; x++, i++) {
+            if (run[i] == true) {
+                global_domain->locations[i].io_idx = i;
+                global_domain->locations[i].global_idx = j;
+                j++;
+            }
+        }
```

Comment:
again, we don't need to loop over x and y.  That's why we added `ncells_active` and `ncells_total`.